cfme/middleware/provider/__init__.py

```diff
@@ -466,34 +379,31 @@ def undeploy(self):
         """
         Clicks on "Undeploy" menu item and verifies message shown
         """
-        self.load_details()
-        operations_btn("Undeploy", invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_success_message('Undeployment initiated for selected deployment(s)')
+        view = self.load_details()
+        view.toolbar.operations.item_select("Undeploy", handle_alert=True)
+        view.flash \
+            .assert_success_message('Undeployment initiated for selected deployment(s)')
 
     def restart(self):
         """
         Clicks on "Restart" menu item and verifies message shown
         """
-        self.load_details()
-        operations_btn("Restart", invokes_alert=True)
-        sel.handle_alert()
-        flash.assert_success_message('Restart initiated for selected deployment(s)')
+        view = self.load_details()
```

Comment:
honestly I don't think load_details() method is necessary at all. it does almost the same as navigate_to(obj, 'Details')

Reply:
Here 'Deployable' is a superclass, and load_details() method is overridden in subclasses with necessary calls to "navigate_to()" and other specific calls.
In general, I keep a style to call "navigate_to()" only from "load_details()" method of particular class, for the rest self.load_details() is called.